target/hexagon/idef-parser/idef-parser.y

```diff
@@ -143,6 +145,101 @@ var : VAR
       }
     ;
 
+/*
+ * Here the integer types are defined from valid combinations of
+ * "signed", "unsigned", "int", and "long" tokens.  The "signed"
+ * and "unsigned" tokens are here assumed to always be placed
+ * first in the type declaration, which is not the case in
+ * normal C. Similarly, "int" is assumed to be placed in the type.
+ */
+type_int : TYPE_INT
+         | TYPE_SIGNED
+         | TYPE_SIGNED TYPE_INT;
+type_uint : TYPE_UNSIGNED
+          | TYPE_UNSIGNED TYPE_INT;
+type_long : TYPE_LONG
+          | TYPE_LONG TYPE_INT
+          | TYPE_SIGNED TYPE_LONG
+          | TYPE_SIGNED TYPE_LONG TYPE_INT;
+type_ulong : TYPE_UNSIGNED TYPE_LONG
+           | TYPE_UNSIGNED TYPE_LONG TYPE_INT;
+type_longlong : TYPE_LONG TYPE_LONG
+              | TYPE_LONG TYPE_LONG TYPE_INT
+              | TYPE_SIGNED TYPE_LONG TYPE_LONG
+              | TYPE_SIGNED TYPE_LONG TYPE_LONG TYPE_INT;
+type_ulonglong : TYPE_UNSIGNED TYPE_LONG TYPE_LONG
+               | TYPE_UNSIGNED TYPE_LONG TYPE_LONG TYPE_INT;
```

Comment:
It seems to me that all of these could be avoided if we were to anticipate some of this logic in the parser.

Reply:
Hmm I'm not sure I understand what you mean here. Do you mean in the lexer by for instance having a general `TYPE_INT` token and then lexing or example
```c
"unsigned long long"    { yylval->rvalue.signedness = UNSIGNED;
                                       yylval->rvalue.bit_width = 64;
                                       return TYPE_INT; }
```
Implementing the int types like this would also work, I just wanted to avoid hard coding the spaces. I believe the only int types that appear in the input are `int`, `unsigned int`, `unsigned long long`, and the`size_t`'s. All of these could be covered this way by a single `TYPE_INT` token so that would be nice.

 Or did you have something else in mind? 

Lastly, should we support only the int types in the input file, or should we include other combinations such as "signed long" and so on to be more future proof?

Reply:
Mh, I see what you mean with the spaces things.
It's a bit verbose, but it makes sense.